### PR TITLE
Re-Upload of "Small patch that addresses the "reduce number of requests sent" issue"

### PR DESCRIPTION
**Pull request by The-Glit-ch**
Re-Upload due to my branch not being up to data with current branch

(#44)

### CODE REVIEW

**Comment by AnonymouX47** [comment]
This is can be merged bu the commit history is still not the best.
What i was suggesting earlier is quite different... sorry if i caused any misunderstanding.

I checked the `main` branch of [your fork](https://github.com/The-Glit-ch/SpotiByeAds) and it was already inconsistent with the upstream's `main` branch which was caused by your earlier commit https://github.com/The-Glit-ch/SpotiByeAds/commit/798bdfe35a41ab0cf63807472107c83bcac27cc7.

Merging this PR into `main` branch of https://github.com/daspartho/SpotiByeAds could be catastrophic, it'll lead to inconsistencies in commit histories between upstream's `main` and the `main` of every fork.

The solution in a case like this would normally be a rebase of your fork's `main` branch (which is technically still possible but could be complicated and messy).
The simplest solution now is to go back and do this the proper way i.e:
1. Close this PR.
2. Delete **your own fork** of the repo (from it's "settings").
3. Create a new fork,
4. Create a new branch (i.e in your fork).
5. Make the changes.
6. Open a PR requesting to merge the new branch (having the change) with upstream's `main`.
7. Once the PR is merged, you can delete the new branch,
8. Fetch from upstream's `main` and merge into your fork's `main`.
9. Voila! Problem fixed.

Sorry for any stress caused, it's all for the better... I could make the change but it was your idea and it's normal not leave it to you to make the PR.
Thanks

File: main.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 #Vars
 keyboard = Controller()
+SubtractDelay = 6 #Check ln 73
```
Comment (AnonymouX47) [inline main.py:15]: This is not required the value can simply be put as a literal constant on line 73 since it's only used once.

File: main.py [hunk 2]
```diff
@@ -69,7 +70,10 @@ def main(username, scope, clientID, clientSecret, redirectURI, path):
         except TypeError:
             pass
         
-        time.sleep(1)
+        time.sleep((spotify.current_user_playing_track['duration_ms']/1000) - SubtractDelay) #Stop us from getting rate limited from spotify API
+        #Instead of checking every second for an ad. We simply get the total length of the song and wait for that ammount of time. 
+        #Now spotify returns miliseconds for song length which is weird but this can easily be fixed with some basic math.
+        #We get the song duration and divide it by 1000 and we get our time in seconds
```
Comment (AnonymouX47) [inline main.py:76]: These are unnecessary, the comment on line 73 is enough (it explains WHY, the code already explains HOW and the names are explicit enough).

File: main.py [hunk 3]
```diff
@@ -69,7 +70,10 @@ def main(username, scope, clientID, clientSecret, redirectURI, path):
         except TypeError:
             pass
         
-        time.sleep(1)
+        time.sleep((spotify.current_user_playing_track['duration_ms']/1000) - SubtractDelay) #Stop us from getting rate limited from spotify API
```
Comment (AnonymouX47) [inline main.py:73]: Maybe make the comment here to be on a separate line to reduce the line length.

**Review by AnonymouX47 — COMMENTED** [review]
Please take not of these comments when opening a new PR.

**Comment by AnonymouX47** [comment]
@The-Glit-ch Just calling your attention to the comments.

**Comment by The-Glit-ch** [comment]
Alright gonna go do that

**Comment by AnonymouX47** [comment]
Also, you should probable open the new PR as a draft :)

**Pull request closed without merging**